src/ProtoRoute/Message.hs

```diff
@@ -1,26 +1,35 @@
 module ProtoRoute.Message
-    ( constructProtoMessage
-    , MessageName
-    , FieldName
-    , FieldValue
-    ) where
+    ( MsgName (..)
+    , FieldName (..)
+    , FVType (..)
+    , Msg (..)
+    , FieldValue (..)
+    , constructProtoMsg) where
 
-import Data.ProtoLens
-import Data.Text (pack)
-import ProtoExports
+import Data.Text (pack, Text)
 
+newtype MsgName   = MN Text deriving (Show)
+newtype FieldName = FN Text deriving (Show)
 
-newtype MessageName = MessageName { unMessageName :: Text }
-newtype FieldName   = FieldName   { unFieldName   :: Text }
-data    FieldValue  = FV [Int] | FV [Text] | FV [FieldValue]
+data FieldValue a = Req a | Opt (Maybe a) | Rep [a] deriving (Show)
+data FVType       = FieldText (FieldValue Text) | FieldInt (FieldValue Int) |
+                    FieldMsg (FieldValue Msg) deriving (Show)
+-- Typesafe representation of a protobuf mesage
+data Msg = Msg
```

Comment:
No need to shorten this constructor and type name to `Msg`. You can call it `Message`.